lib/std/fmt.zig

```diff
@@ -1125,6 +1125,90 @@ pub fn formatIntBuf(out_buf: []u8, value: anytype, base: u8, uppercase: bool, op
     return fbs.pos;
 }
 
+/// Formats a duration in nanoseconds according to its magnitude:
+///
+/// - #ns
+/// - #[.#]us
+/// - #[.#]ms
+/// - #[.#]s
+/// - #m#[.#]s
+/// - #h#m#[.#]s
+/// - #d#h#m#[.#]s
+/// - #w#d#h#m#[.#]s
+pub fn formatDuration(ns: u64, writer: anytype) !void {
+    var ns_remaining = ns;
+
+    inline for (.{
+        .{ .ns = std.time.ns_per_week, .sep = 'w' },
```

Comment:
add year as well?

We're so close to prometheus' https://prometheus.io/docs/prometheus/latest/querying/basics/#time-durations

> 
> Time durations are specified as a number, followed immediately by one of the following units:
> 
>  -  ms - milliseconds
>  -  s - seconds
>  -  m - minutes
>  -  h - hours
>  -  d - days - assuming a day has always 24h
>  -  w - weeks - assuming a week has always 7d
>  -  y - years - assuming a year has always 365d
> 
> Time durations can be combined, by concatenation. Units must be ordered from the longest to the shortest. A given unit must only appear once in a time duration.
> 
> Here are some examples of valid time durations:
> 
> - 5h
> - 1h30m
> - 5m
> - 10s

Reply:
Can do!